app-frontend/src/app/components/pages/project/layer/aoi/index.js

```diff
@@ -1,11 +1,286 @@
+    fetchPage(page = 1, typeId = this.filterList[0].id) {
+        this.itemList = [];
+        let fetchedList = [];
+        this.dataType = this.filterList.find(fl => fl.id === typeId).type;
+        const query = this.getQueryByType(page);
+
+        const currentQuery = query.then((paginatedResponse) => {
+            if (this.dataType === 'AOI') {
+                fetchedList = paginatedResponse.results;
+            } else {
+                fetchedList = paginatedResponse.features;
+            }
+
+            this.itemList = _.compact(fetchedList.map(fl => {
+                if (this.dataType === 'AOI' && !fl.geometry) {
+                    return 0;
+                }
+                return this.createItemInfo(fl);
+            }));
+
+            if (this.itemList.length && this.dataType === 'AOI') {
+                this.onSelect(this.layer.id);
+            } else if (this.itemList.length && this.dataType === 'VECTOR') {
+                this.onSelect(this.itemList[0].id);
+            }
+
+            this.pagination = this.paginationService.buildPagination(paginatedResponse);
+            this.paginationService.updatePageParam(page);
+            if (this.currentQuery === currentQuery) {
+                delete this.fetchError;
+            }
+        }, (e) => {
+            if (this.currentQuery === currentQuery) {
+                this.fetchError = e;
+            }
+        }).finally(() => {
+            if (this.currentQuery === currentQuery) {
+                delete this.currentQuery;
+            }
+        });
+        this.currentQuery = currentQuery;
+        return currentQuery;
+    }
+
+    getQueryByType(page) {
+        return this.dataType === 'AOI' ?
+            this.projectService.getProjectLayers(this.project.id,
+                {pageSize: 30, page: page - 1}) :
+            this.shapesService.query({pageSize: 30, page: page - 1});
+    }
+
+    onConfirmAoi(aoiGeojson, isSaveShape) {
+        const geom = this.getMultipolygonGeom(aoiGeojson);
+        if (!geom) {
+            this.$window.alert('The supplied AOI geometry is incorrect. Please try again.');
+            return;
+        }
+        const layerGeomToUpdate = this.generateFeature(geom);
+        const updatedLayer = Object.assign({}, this.layer, {geometry: layerGeomToUpdate});
+        this.projectService
+            .updateProjectLayer(updatedLayer)
+            .then(newLayer => {
+                this.layer = newLayer;
+                this.layerGeom = this.layer.geometry;
+                if (isSaveShape) {
+                    this.openShapeModal(this.layer);
+                } else {
+                    this.$state.go('project.layer');
+                }
+            })
+            .catch(err => {
+                this.$log.error(err);
+                this.$window.alert('There is an error updating this layer\'s AOI.'
+                    + ' Please try again later');
+            });
+    }
+
+    getMultipolygonGeom(aoiGeojson) {
```

Comment:
I'd do `const geomType = _.get(aoiGeojson, 'geometry.type')` to get around the long conditionals.